atc/syslog/drainer.go

```diff
@@ -115,3 +102,144 @@ func (d *drainer) drainBuild(logger lager.Logger, build db.Build, syslog *Syslog
+		if err != nil {
+			logger.Error("failed-to-unmarshal", err)
+			return err
+		}
+		ts = time.Unix(initEvent.Time, 0)
+		tag = build.SyslogTag(initEvent.Origin.ID)
+		message = fmt.Sprintf("initializing")
+	case event.EventTypeInitializeGet:
+		var initGetEvent event.InitializeGet
+		err := json.Unmarshal(*ev.Data, &initGetEvent)
+		if err != nil {
+			logger.Error("failed-to-unmarshal", err)
+			return err
+		}
+		ts = time.Unix(initGetEvent.Time, 0)
+		tag = build.SyslogTag(initGetEvent.Origin.ID)
+		message = fmt.Sprintf("get initializing")
+	case event.EventTypeInitializePut:
+		var initPutEvent event.InitializePut
+		err := json.Unmarshal(*ev.Data, &initPutEvent)
+		if err != nil {
+			logger.Error("failed-to-unmarshal", err)
+			return err
+		}
+		ts = time.Unix(initPutEvent.Time, 0)
+		tag = build.SyslogTag(initPutEvent.Origin.ID)
+		message = fmt.Sprintf("put initializing")
+	case event.EventTypeInitializeTask:
+		var initTaskEvent event.InitializeTask
+		err := json.Unmarshal(*ev.Data, &initTaskEvent)
+		if err != nil {
+			logger.Error("failed-to-unmarshal", err)
+			return err
+		}
+		ts = time.Unix(initTaskEvent.Time, 0)
+		tag = build.SyslogTag(initTaskEvent.Origin.ID)
+		message = fmt.Sprintf("task initializing")
+	case event.EventTypeSelectedWorker:
+		var selectedWorkerEvent event.SelectedWorker
+		err := json.Unmarshal(*ev.Data, &selectedWorkerEvent)
+		if err != nil {
+			logger.Error("failed-to-unmarshal", err)
+			return err
+		}
+		ts = time.Unix(selectedWorkerEvent.Time, 0)
+		tag = build.SyslogTag(selectedWorkerEvent.Origin.ID)
+		message = fmt.Sprintf("selected worker: %s", selectedWorkerEvent.WorkerName)
+	case event.EventTypeStartTask:
+		var startTaskEvent event.StartTask
+		err := json.Unmarshal(*ev.Data, &startTaskEvent)
+		if err != nil {
+			logger.Error("failed-to-unmarshal", err)
+			return err
+		}
+		ts = time.Unix(startTaskEvent.Time, 0)
+		tag = build.SyslogTag(startTaskEvent.Origin.ID)
+
+		buildConfig := startTaskEvent.TaskConfig
+		argv := strings.Join(append([]string{buildConfig.Run.Path}, buildConfig.Run.Args...), " ")
+		message = fmt.Sprintf("running %s", argv)
+	case event.EventTypeLog:
+		var logEvent event.Log
+		err := json.Unmarshal(*ev.Data, &logEvent)
+		if err != nil {
+			logger.Error("failed-to-unmarshal", err)
+			return err
+		}
+		ts = time.Unix(logEvent.Time, 0)
+		tag = build.SyslogTag(logEvent.Origin.ID)
+		message = logEvent.Payload
+	case event.EventTypeFinishGet:
+		var finishGetEvent event.FinishGet
+		err := json.Unmarshal(*ev.Data, &finishGetEvent)
+		if err != nil {
+			logger.Error("failed-to-unmarshal", err)
+			return err
+		}
+		ts = time.Unix(finishGetEvent.Time, 0)
+		tag = build.SyslogTag(finishGetEvent.Origin.ID)
+
+		version, _ := json.Marshal(finishGetEvent.FetchedVersion)
+		metadata, _ := json.Marshal(finishGetEvent.FetchedMetadata)
+		message = fmt.Sprintf("{\"version\": %s, \"metadata\": %s", string(version), string(metadata))
```

Comment:
We should differentiate between `get` and `put` events

```suggestion
		message = fmt.Sprintf("get {\"version\": %s, \"metadata\": %s", string(version), string(metadata))
```

Reply:
Sure, updated.